composer/core/state.py

```diff
@@ -292,6 +295,8 @@ def __init__(
         dataloader: Optional[Iterable] = None,
         dataloader_label: Optional[str] = None,
         dataloader_len: Union[int, Time[int]] = -1,
+        dataset_state: Optional[Dict[str, Any]] = None,
+        dataset_resumption: Optional[Dict[str, Any]] = None,
```

Comment:
Needs to be added to doc strings above. We should definitely write a good bit on what each of these are doing because the logic here is subtle and somewhat hard to understand -- it took me a while to remember everything and I helped design the original version :P

Reply:
added docstrings throughout